designs/node-repair.md

```diff
@@ -0,0 +1,182 @@
+# Node Auto Repair 
+
+## Problem Statement
+
+Nodes can experience failure modes that cause degradation to the underlying hardware, filesystem, or container environment. Some of these failure modes are surfaced through the Node object (consider referencing some of them) while others are not surfaced at all (consider referencing some of these). A Diagnostic Agent such as [Node Problem Detector(NPD)](https://github.com/kubernetes/node-problem-detector) offers a way to surface these failures as additional status conditions on the node object.
+
+In either case, even if a status condition is surfaced through the Node that indicates that the Node is unhealthy, Karpenter doesn't currently react to this unhealthiness in any way today.
+
+* Mega Issue: https://github.com/kubernetes-sigs/karpenter/issues/750
+    * Related (Unreachable): https://github.com/aws/karpenter-provider-aws/issues/2570
+    * Related (Remove by taints): https://github.com/aws/karpenter-provider-aws/issues/2544
+    * Related (Known resource are not registered) Fixed by v0.28.0: https://github.com/aws/karpenter-provider-aws/issues/3794
+    * Related (Stuck on NotReady): https://github.com/aws/karpenter-provider-aws/issues/2439
+
+#### Out of scope
+
+The alpha implementation will not consider disruption budgets, or API surface for repairing unhealthy nodes. The team does not have enough data to determine the right level of configuration surface that users will utilize. The goal will be to provide an opinionated mechanism help customer not need to consider the definition of unhealthy nodes, ultimately reducing configuration burden for most customers. **The feature will be gated under an alpha NodeRepair=true feature flag to allow customer to enable the feature. This will  allow for additional feedback from customers, and make subsequent changes in the future that can support the features we have considered out of scope for the Alpha stage.**
```

Comment:
nit: I'd consider removing the prose and just use bullet points to say, "Here's the things that we aren't considering for this iteration"

- Disruption Budgets
- Customer-Defined Conditions
- Customer-Defined Remediation Time